gerrychain/partition/partition.py

```diff
@@ -51,6 +55,11 @@ def _first_time(self, graph, assignment, updaters):
         self.flows = None
         self.edge_flows = None
 
+    def validate_assignment(self):
+        node_names = set(list(self.graph.nodes()))
+        assgn_names = set([name for dist in self.assignment.parts.values() for name in dist])
```

Comment:
The `set()` constructor can take the plain generator expression instead of the list:
```suggestion
        assgn_names = set(name for dist in self.assignment.parts.values() for name in dist)
```